File: Lab_2/task_3/3.cs
using System;

class Program
{
    // Функция для переворачивания числа и удаления ведущих нулей
    static int ReverseNumber(int num)
    {
        int reversed = 0;
        while (num > 0)
        {
            reversed = reversed * 10 + num % 10;
            num /= 10;
        }
        return reversed;
    }

    static void Main()
    {
        Console.Write("Введите количество чисел: ");
        int n = int.Parse(Console.ReadLine());

        Console.Write($"Введите {n} чисел: ");
        for (int i = 0; i < n; ++i)
        {
            int num = int.Parse(Console.ReadLine());
            Console.Write(ReverseNumber(num) + " ");
        }

        Console.WriteLine();
    }
}

File: Lab_2/task_3/3.rs
use std::io;

fn reverse_number(mut num: i32) -> i32 {
    let mut reversed = 0;
    while num > 0 {
        reversed = reversed * 10 + num % 10;
        num /= 10;
    }
    reversed
}

fn main() {
    let mut input = String::new();
    println!("Введите количество чисел:");
    io::stdin().read_line(&mut input).unwrap();
    let n: i32 = input.trim().parse().unwrap();

    println!("Введите {} чисел:", n);
    for _ in 0..n {
        let mut num_input = String::new();
        io::stdin().read_line(&mut num_input).unwrap();
        let num: i32 = num_input.trim().parse().unwrap();
        print!("{} ", reverse_number(num));
    }

    println!();
}
